src/View/mainpage/DrawROIWindow/UIDrawROIWindow.py

```diff
@@ -862,29 +880,146 @@ def onFillClicked(self):
+            for x in range(0, 2):
+                print("x == " + str(x))
+                if x == 0:
+                    range_start = id - 1
+                    range_end = 1
+                    step = -1
+                else:
+                    range_start = id
+                    range_end = len(self.patient_dict_container.dataset) - 1
+                    step = 1
+                # Search the slides in the above ranges (down and up)
+                for y_search in range(range_start, range_end, step):
+                    print(str(range_start) + " range end: " + str(range_end) + " range: " + str(step))
+                    temp_id = y_search
+
+                    self.dicom_view.slider.setValue(temp_id)
+
+                    dt = self.patient_dict_container.dataset[temp_id]
+                    dt.convert_pixel_data()
+
+                    # Path to the selected .dcm file
+                    location = self.patient_dict_container.filepaths[temp_id]
+                    self.ds = pydicom.dcmread(location)
+
+                    self.drawingROI = Drawing(
+                        pixmaps[temp_id],
+                        dt._pixel_array.transpose(),
+                        min_pixel,
+                        max_pixel,
+                        self.patient_dict_container.dataset[temp_id],
+                        self.draw_roi_window_instance,
+                        self.slice_changed,
+                        self.current_slice,
+                        self.drawing_tool_radius,
+                        self.keep_empty_pixel,
+                        self.pixel_transparency,
+                        set(),
+                        xy=self.seed
+                    )
+                    print("xy = " + str(self.seed))
+                    self.slice_changed = True
+                    self.has_drawing = True
+                    self.dicom_view.view.setScene(self.drawingROI)
+                    self.enable_cursor_diameter_change_box()
+
+                    if self.drawingROI._display_pixel_color() != True:
+                        break
+        else:
+            dt = self.patient_dict_container.dataset[id]
+            dt.convert_pixel_data()
+            # Path to the selected .dcm file
+            location = self.patient_dict_container.filepaths[id]
+            self.ds = pydicom.dcmread(location)
+
+            self.drawingROI = Drawing(
+                pixmaps[id],
+                dt._pixel_array.transpose(),
+                min_pixel,
+                max_pixel,
+                self.patient_dict_container.dataset[id],
+                self.draw_roi_window_instance,
+                self.slice_changed,
+                self.current_slice,
+                self.drawing_tool_radius,
+                self.keep_empty_pixel,
+                self.pixel_transparency,
+                set(),
+                UI=self
+            )
+
+            self.slice_changed = True
+            self.has_drawing = True
```

Comment:
I don't think I fully understand but this is repeated. It should be taken out into another method with the variables that actually change

Reply:
We need to have a call about this